### PR TITLE
Feature/convert vdypsi32 to java

### PR DESCRIPTION
**Pull request by mjunkin**

### CODE REVIEW

**Review by smithkm — APPROVED** [review]
Looks good.  I just had a few comments, questions, and minor nit picks.

File: vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/common_calculators/BaseAreaTreeDensityDiameter.java [hunk 1]
```diff
@@ -2,28 +2,48 @@
...
-	private BaseAreaTreeDensityDiameter() {
-	}
-
+	private BaseAreaTreeDensityDiameter() {}
+	
 	/**
 	 * π/10⁴
 	 */
-	public static final float PI_40K = (float) (Math.PI / 40_000);
+	public static final float PI_40K = (float) (Math.PI / 40000);
```
[the remark this answers is not in the document]
Comment (mjunkin) [reply vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/common_calculators/BaseAreaTreeDensityDiameter.java:15]: No - I'd just never seen that syntax before, was confused, and thought others may be. I'll change it back however.

File: vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/common_calculators/enumerations/SiteIndexForestInventoryZone.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package ca.bc.gov.nrs.vdyp.common_calculators.enumerations;
+
+public enum SiteIndexForestInventoryZone {
```
[the remark this answers is not in the document]
Comment (mjunkin) [reply vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/common_calculators/enumerations/SiteIndexForestInventoryZone.java:3]: "SiteIndex" is just a prefix for enumerations pulled from sindex33. Yes, it's the same as Region, but I wanted to keep the common_calculators code logically isolated given its origin.

File: vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/common_calculators/enumerations/SiteIndexSpecies.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+	public String getCode() {
+		return code;
+	}
+	
+	public int n() {
+		return n;
+	}
+	
+	public static SiteIndexSpecies getFirstSpecies() {
+		return SI_SPEC_A;
```
[the remark this answers is not in the document]
Comment (mjunkin) [reply vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/common_calculators/enumerations/SiteIndexSpecies.java:166]: OK, done.

File: vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/common_calculators/Age2Age.java [hunk 4]
```diff
@@ -1,259 +1,198 @@
...
-	private static final int SI_SW_GOUDNIGH = 85;
-	private static final int SI_SW_NIGHTA = 83;
-	/* not used, but must be defined for array positioning */
-	private static final int SI_SE_NIGHTA = 110;
-	private static final int SI_SW_NIGHTA2004 = 111;
-	private static final int SI_PLI_NIGHTA2004 = 109;
-
-	public static double age_to_age(short cu_index, double age1, short age1_type, short age2_type, double y2bh) {
+	@SuppressWarnings("java:S3776, java:S6541, java:S1479")
+	public static double ageToAge(SiteIndexEquation cuIndex, double sourceAge, SiteIndexAgeType sourceAgeType, SiteIndexAgeType targetAgeType, double years2BreastHeight)
```
[the remark this answers is not in the document]
Comment (mjunkin) [reply vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/common_calculators/Age2Age.java]: ok

File: vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/model/GenusDistribution.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	private final GenusDefinition genus;
+	private final float percentage;
+
+	/**
+	 * Construct a GenusDistribution.
+	 * @param index the genus's index.
+	 * @param genus the genus. Must not be null.
+	 * @param percentage the distribution percentage. Must be non null and in the range 0 .. 100.
+	 */
+	public GenusDistribution(int index, GenusDefinition genus, Float percentage) {
```
[the remark this answers is not in the document]
Comment (mjunkin) [reply vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/model/GenusDistribution.java:19]: So if it is null we get to intervene and produce a nicer error message. If it accepted float instead, auto-unboxing would result in an NPE being thrown...

File: vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/model/UtilizationClass.java [hunk 6]
```diff
@@ -18,6 +22,9 @@ public enum UtilizationClass {
 	private Optional<UtilizationClass> next = Optional.empty();
 	private Optional<UtilizationClass> previous = Optional.empty();
 
+	public static final List<UtilizationClass> allButSmall = Collections
```
[the remark this answers is not in the document]
Comment (mjunkin) [reply vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/model/UtilizationClass.java:25]: Sad but true. I've changed it.

File: vdyp-si32/src/main/java/ca/bc/gov/nrs/vdyp/si32/enumerations/SpeciesRegion.java
[the remark this answers is not in the document]
Comment (mjunkin) [reply vdyp-si32/src/main/java/ca/bc/gov/nrs/vdyp/si32/enumerations/SpeciesRegion.java:1]: Same argument as above.

File: vdyp-forward/src/main/java/ca/bc/gov/nrs/vdyp/forward/model/VdypPolygonDescription.java
[the remark this answers is not in the document]
Comment (mjunkin) [reply vdyp-forward/src/main/java/ca/bc/gov/nrs/vdyp/forward/model/VdypPolygonDescription.java:1]: Yes. I'll leave it to later.

**Pull request merged**